### ISSUE TITLE
Naming convention for macros internal to the headers

### ISSUE THREAD

**Issue by kainino0x** [webgpu-native/webgpu-headers#355]
Right now we have a few macros internal to headers:

- `__WGPU_EXTEND_ENUM` which the generator produces only in extension headers (not the core `webgpu.h`)
- `WGPU_MAKE_INIT_STRUCT` and `WGPU_COMMA`

What should the naming convention for these be? Started a discussion on #353:

@kainino0x:
> I think libraries are supposed to avoid macro names starting with underscores - see https://gcc.gnu.org/onlinedocs/cpp/System-specific-Predefined-Macros.html
> 
> We could alternatively come up with some other valid naming convention for internal things like _wgpu_EXTEND_ENUM, WDYT?

@rajveermalviya:
> +1 for `_wgpu_EXTEND_ENUM` (or if possible `_WGPU_EXTEND_ENUM`), as it's only intended for internal use.

We can't use `__*` or `_WGPU*` because those are reserved but `_wgpu*` or `_internal_WGPU*` would be fine.

Optionally (regardless of the naming convention) we could `#undef` them at the end of the file; this seems safe to me though it prevents us from defining anything in `webgpu.h` and using it in an extension header.
If we undef then it would probably be fine if we just used `WGPU*` without an "internal" indicator.

**Comment by kainino0x** [comment]
cc @Kangz

**Comment by Kangz** [comment]
No strong opinion, `_wgpu_*` and leaving them defined at the end for extension headers?

**Comment by kainino0x** [comment]
I found one random thing online (lost where it was) that indicated `_wgpu_*` might not be valid either, so I dug a bit.

**tl;dr: I don't know, but most people on the internet seem to agree with `clang -Weverything`, which says `_wgpu_*` is most likely fine.**

I tried looking at the [C17 spec](https://web.archive.org/web/20181230041359if_/http://www.open-std.org/jtc1/sc22/wg14/www/abq/c17_updated_proposed_fdis.pdf) [7.1.3]:

> - All identifiers that begin with an underscore and either an uppercase letter [`_A*`] or another underscore [`__*`] are always reserved for any use, except those identifiers which are lexically identical to keywords.187)
> - All identifiers that begin with an underscore [`_*`] are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces.

but it's hard to understand - which name space are macros in? (They say they have their own name space [6.10.3.7], but it's not listed in the list of namespaces [6.2.3.1].)

Clang seems to think that `_a` is invalid for global (file-scope) variables and structs, but valid for macros and local variables:
https://godbolt.org/z/j6bKG7zhW
But are macros file-scope? I don't know.

Similarly libc says `_a` is invalid for ["external identifiers"](https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html) but I don't know what external means (external to header or external to the compilation unit - i.e. linkable things?)

As an aside, I also found out that macOS clang is forcing a bunch of warnings off even when I use `-Weverything`, including this one (visible via `clang -v`), so I'm going to force those back on in the compile tests.
